### PR TITLE
feat(cli): use hidden prompt or stdin for auth:token command

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/CDX-665

## Proposed changes

use hidden prompt or stdin for auth:token command

## Breaking changes

deprecate --token flag and rely on hidden prompt or stdin for auth:token command. default to stdin for ci environment

## Testing

- [x] Unit Tests:

- [ ] Functionnal Tests:

- [x] Manual Tests:

### CODE REVIEW

File: packages/cli/core/src/commands/auth/token.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+import {CliUx, Flags} from '@oclif/core';
 import {Config} from '@coveo/cli-commons/config/config';
 import {AuthenticatedClient} from '@coveo/cli-commons/platform/authenticatedClient';
 import {PlatformEnvironment} from '@coveo/cli-commons/platform/environment';
 import {Region} from '@coveo/platform-client';
 import {withEnvironment, withRegion} from '../../lib/flags/platformCommonFlags';
 import {Trackable} from '@coveo/cli-commons/preconditions/trackable';
 import {formatOrgId} from '@coveo/cli-commons/utils/ux';
+import {readFile} from 'fs';
+import {} from 'readline';
```
Comment (louis-bompart) [inline packages/cli/core/src/commands/auth/token.ts:11]:
```suggestion
```

**Review by dmbrooke — APPROVED** [review]

**Review by dmbrooke — COMMENTED** [review]
Whoops, didn't mean to approve yet. Just started reviewing now 😅

File: packages/cli/core/src/commands/auth/token.ts [hunk 2]
```diff
@@ -18,12 +19,11 @@ export default class Token extends CLICommand {
...
     ...withEnvironment(),
-    token: Flags.string({
-      char: 't',
+    stdin: Flags.boolean({
+      char: 's',
       description:
-        'The API-Key that shall be used to authenticate you to the organization. See <https://github.com/coveo/cli/wiki/Using-the-CLI-using-an-API-Key>.',
-      required: true,
-      helpValue: 'xxx-api-key',
+        'Read the token from stdin. Default to true when running in a CI environment.',
```
Comment (dmbrooke) [inline packages/cli/core/src/commands/auth/token.ts:25]:
```suggestion
        'Whether to read the token from stdin. Default to true when running in a CI environment.',
```

**Review by dmbrooke — APPROVED** [review]
Looks good and seems to be working as expected 🙂

**Pull request merged**